samples/java-spring-choreography-saga-quickstart/README.md

```diff
@@ -0,0 +1,189 @@
+but may also have an email address that needs to be unique across all users in the system.
+
+A **Choreography Saga** can be implemented as a solution to this challenge. In addition to the `UserEntity`, we can 
+implement a second entity to act as a barrier. This entity, called the `UniqueEmailEntity`, will be responsible for ensuring that each email address is associated with only one user.  The unique ID of the `UniqueEmailEntity` will be the email address itself. Thus, we can guarantee that there is only one instance of this entity per email.
+
+When a request to create a new `UserEntity` is received, we first check if the email address is in use by first 
+trying to reserve it using the barrier, the `UniqueEmailEntity`. If it's not already in use, we proceed to create the `UserEntity`. Once the `UserEntity` has been created, the `UniqueEmailEntity` is marked as CONFIRMED.
+
+If the email address is already in use, the request to create the `UserEntity` will fail.
+
+### Successful Scenario
+
+The sunny day scenario is illustrated in the following diagram:
+
+![Successful Flow](flow-successful.png?raw=true)
+
+All incoming requests are handled by the `ApplicationController` which is implemented using a Kalix Action. 
+
+1. upon receiving a request to create a new User, the `ApplicationController` will first reserve the email.
+2. it will then create the User. 
+   1. the `UserEventsSubscriber` Action listens to the User's event 
+   2. and mark the UniqueEmail entity as soon as it 'sees' that a User has been created.
+      
+### Failure Scenario
+
+As these are two independent transactions, we also need to consider the failure scenarios. For example, a request to 
+reserve the email address may succeed, but the request to create the user may fail. In such a scenario, we end up with 
+an email address that is reserved but not associated with a user.
+
+The failure scenario is illustrated in the following diagram:
+
+![Failure Flow](flow-failure.png?raw=true)
+
+1. upon receiving a request to create a new User, the `ApplicationController` will first reserve the email.
+   1. in the background, the `UniqueEmailSubscriber` Action listens to state changes from `UniqueEmailEntity`
+   2. when it detects that an email was reserved it schedules a timer to un-reserve it after a certain amount of time. 
+   3. when the timer fires, the email is un-reserved if the `UniqueEmailEntity` is still in RESERVED state.
+2. then it tries to create the User, but it fails. As such, the email will never be confirmed, but the timer 
+   will unlock it.
+
+> [!NOTE]
+> Everything on the side of the `UniqueEmailSubscriber` is happening in the background and independent of the success or failure of the User creation.
+
+### Full Successful Scenario
+
+Now that we have covered the failure scenario, we can have a look at the full picture for the successful scenario:
+
+![Full Successful Flow](flow-full.png?raw=true)
```

Comment:
Here we are printing the whole flow. In the previous example, we have the explanation of the steps in markdown. And substep (sub-lists) are represented with roman numerals. 

Here I'm not using markdown and then it make it easier to keep arabic numerals. Otherwise, we will have twice `I`.

Reply:
The problem here is also that you can not create sub-list in markdown using 2.1, 2.2, etc. If it was possible, it would make our life easy.

Reply:
I will revisit the usage of `I`, `II` and `III`. I was also not so happy with it, but couldn't find a better way of expressing it in drawing and in text.

Reply:
Revisited.